cmd/krew/cmd/list.go

```diff
@@ -56,20 +134,56 @@ Remarks:
 				return nil
 			}
 
-			// print table
-			var rows [][]string
-			for p, version := range plugins {
-				rows = append(rows, []string{p, version})
+			if *output.OutputFormat == "wide" || (*output.OutputFormat == "" && isTerminal(os.Stdout)) {
+				// print table
+				var rows [][]string
+				for p, version := range plugins {
+					rows = append(rows, []string{p, version})
+				}
+				rows = sortByFirstColumn(rows)
+				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
+			}
+
+			if sliceContains(*output.OutputFormat, output.AllowedFormats()) {
+				objs := make([]Entry, 0, len(plugins))
+				for plugin, version := range plugins {
+					obj := Entry{plugin, version}
+					objs = append(objs, obj)
+				}
+				objs = sortByName(objs)
+				for _, obj := range objs {
+					p, err := output.ToPrinter()
+					if err != nil {
+						return err
+					}
+					err = p.PrintObj(obj, os.Stdout)
+					if err != nil {
+						return err
+					}
+				}
+				return nil
 			}
-			rows = sortByFirstColumn(rows)
-			return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
+
+			return errors.New("unsupported output format specified")
 		},
 		PreRunE: checkIndex,
 	}
 
+	listCmd.Flags().StringVarP(&outputFormat, "output", "o", "", "_")
```

Comment:
Let's add a description (unless "_" is a magic string that makes it work?)

I'm not sure if genericclioptions help here.

Reply:
I'll commit this description, hope that's okay. `Output format. One of json, yaml or wide`